lambdas/observe/observe_lambda_function.py

```diff
@@ -62,10 +71,40 @@ def lambda_handler(event, context):
             # Update the marker in DynamoDB
             data_service.update_marker(marker)
             logger.info(f"Successfully updated marker with ID {marker.get_marker_id()}.")
+
+            # Check notification condition and notify subscribers
+            if should_notify(marker):
+                emails = marker.get('subscribedEmails', [])
+                if emails:
+                    try:
+                        notification_service.notify_subscribers(marker.get_marker_id(), emails)
+                        logger.info(f"Notifications sent for marker {marker.get_marker_id()} to {emails}.")
+                    except Exception as e:
+                        logger.error(f"Failed to notify subscribers for marker {marker.get_marker_id()}: {e}")
+                else:
+                    logger.info(f"No subscribers found for marker {marker.get_marker_id()}.")
```

Comment:
@murauskas101 create and move this into another for loop. After this `for marker in markers` create another one `for marker in markers` that'll just handle notifications. The way we have separation of concerns. One loop does updates. Another does reads and notifications